docs/rules/no-useless-backreference.md

```diff
@@ -0,0 +1,130 @@
+# Disallow useless backreferences in regular expressions (no-useless-backreference)
+
+Backreferences that always successfully match zero-length and cannot match anything else are useless.
```

Comment:
I wonder if a sentence could be added explaining that while JavaScript allows cyclical backreferences, forward references, etc., that they are interpreted as zero-width matches. As it is, starting the documentation with "Backreferences that always [...] are useless" feels like it comes out of nowhere, if the reader isn't familiar with the issue and doesn't know what it means.

Reply:
Thanks!

Completely agree. I wanted the docs to be specific about what this rule targets because there can be some other useless backreference that the rule can't detect, so all text ended up in the Rule Details section.

Added an opening paragraph with more details now.